src/dex-ui-components/base/Inputs/FileUploader.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback } from "react";
+import { useDropzone } from "react-dropzone";
+import { Card, Text } from "@chakra-ui/react";
+import { Color } from "../../../dex-ui-components";
+interface File extends Blob {
+  readonly lastModified: number;
+  readonly name: string;
+  readonly webkitRelativePath: string;
+}
+
+const defaultFileUploaderStyles = {
+  bg: "white",
+  width: "100%",
+  textAlign: "center",
+  justifyContent: "center",
+  alignSelf: "center",
+  border: "1px #31A9BD dashed",
+  height: "102Px",
+};
+interface FileUploaderProps {
+  title: string;
+  subTitle: string;
+  onHoverTitle: string;
+  fileUploaderStyles?: any;
+  disabled?: boolean;
+  onFileRead: (file: string, fileName: string) => void;
+  onFileReadFailed?: (reason: string) => void;
+}
+
+function FileUploader(props: FileUploaderProps) {
+  const { title, subTitle, onHoverTitle, fileUploaderStyles, disabled, onFileRead, onFileReadFailed } = props;
+
+  const onDrop = useCallback(
+    (acceptedFiles: File[]) => {
+      acceptedFiles.forEach((file) => {
+        const reader = new FileReader();
+        reader.onabort = () => {
+          if (onFileReadFailed) onFileReadFailed("File reading has aborted");
+        };
+        reader.onerror = () => {
+          if (onFileReadFailed) onFileReadFailed("File reading has failed");
+        };
+        reader.onload = () => {
+          const dataUrl = reader.result as string;
+          onFileRead(dataUrl, file.name);
+        };
+        reader.readAsText(file);
+      });
+    },
+    [onFileRead, onFileReadFailed]
+  );
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    maxFiles: 1,
+    accept: { "application/json": [".json"] },
+    disabled: disabled ?? false,
+  });
+
+  return (
+    <Card {...getRootProps()} style={{ ...defaultFileUploaderStyles, ...fileUploaderStyles }}>
+      <input {...getInputProps()} />
+      {isDragActive ? (
+        <Text color={Color.Blue_03}>{onHoverTitle}</Text>
+      ) : (
+        <>
+          <Text color={Color.Blue_03}>{title}</Text>
```

Comment:
Use the `textStyle` variants to define text attributes. You can then override attributes like `color` using inline styles (like you are already doing). 

It looks like the wireframes specified this as `body-02` text which maps to `b2` in our code.
![Screen Shot 2022-12-21 at 11 50 15 AM](https://user-images.githubusercontent.com/54907098/208959961-3ba2eaa7-f14f-42bd-ac75-4c54238c586d.png)

`<Text textStyle="b2" color={Color.Blue_03}>{title}</Text>`

The list of available custom text variants extracted from the wireframes is in the **dex-ui-components/base/Text/styles/TextStyles.ts** file.

Note that Typography defined in the wireframes should but may not always map one-to-one with the text variants described in our code. If you run into any discrepancies, you can reach out to myself or Anna.

Reply:
Also, it looks like the wireframes specify that the text for the file uploader is `blue 02` (Color.Blue_02). 
![Screen Shot 2022-12-21 at 11 59 48 AM](https://user-images.githubusercontent.com/54907098/208961937-ab84989c-1cb2-4a28-a297-65593360589d.png)
I would confirm with Anna before changing this to `blue 03`.